### PR TITLE
LPS-203971 - Allow import Object Folder to continue if errors happen on Object Definition import

### PR DESCRIPTION
**Pull request by G-Sa**
Related to : [LPS-203971](https://liferay.atlassian.net/browse/LPS-203971)

### CODE REVIEW

**Comment by G-Sa** [comment]
ci:test:sf

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFolderResourceImpl.java [hunk 1]
```diff
@@ -243,16 +248,28 @@ private void _addObjectFolderResources(
...
+					failedObjectDefinitionNames.add(objectDefinition.getName());
+
+					objectDefinition =
+						objectDefinitionResource.
+							getObjectDefinitionByExternalReferenceCode(
+								objectDefinition.getExternalReferenceCode());
+				}
 
 				serviceBuilderObjectDefinitionIds.remove(
 					objectDefinition.getId());
```
Comment (marcelabc) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFolderResourceImpl.java:275]: What happens if the `objectDefinition` is Null?
Shouldn't it be safer to verify before removing it from the list?

Comment (marcelabc) [reply modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFolderResourceImpl.java:275]: objectDefinitionResource.getObjectDefinitionByExternalReferenceCode() is a Fetch and can return null.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectFolderMVCActionCommand.java [hunk 2]
```diff
@@ -68,11 +71,32 @@ protected void doProcessAction(
...
 
-			JSONObject jsonObject = JSONUtil.put(
-				"title",
-				_language.get(
-					_portal.getHttpServletRequest(actionRequest),
-					"the-object-folder-failed-to-import"));
+			JSONObject jsonObject = null;
+
+			if (exception instanceof
+					ObjectFolderItemObjectDefinitionIdException) {
```
[the remark this answers is not in the document]
Comment (G-Sa) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectFolderMVCActionCommand.java:77]: Here i followed the same pattern as ImportObjectDefinitionMVCActionCommand. Maybe if we do this change here  we would have to do this aproach in a ton of other places

[the remark this answers is not in the document]
Comment (marcelabc) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectFolderMVCActionCommand.java:77]: Thank you for showing me this pattern. 
Okay, we can keep this way :+1:

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Sent to Brian: https://github.com/brianchandotcom/liferay-portal/pull/145603

**Pull request closed without merging**